File: backend/models/prospection.py
import uuid
from sqlalchemy import Column, String, Integer, Float, DateTime, Text
from sqlalchemy.dialects.mssql import UNIQUEIDENTIFIER
from sqlalchemy.sql import func
from data.connections.database import Base


class Prospection(Base):
    __tablename__ = "prospections"
    __table_args__ = {"schema": "prospector"}

    id = Column(UNIQUEIDENTIFIER, primary_key=True, default=uuid.uuid4)
    tenant_id = Column(String(100), nullable=False, default="prospector-qc")
    titulo = Column(String(200), nullable=False)
    segmento = Column(String(200), nullable=False)
    localizacao = Column(String(200))
    quantidade_leads = Column(Integer, nullable=False, default=10)
    prompt_customizado = Column(Text)
    engine_tipo = Column(String(50), nullable=False, default="claude_code")
    status = Column(String(50), nullable=False, default="pendente")
    resultado_raw = Column(Text)
    total_leads_encontrados = Column(Integer, default=0)
    erro = Column(Text)
    arquivo_export_url = Column(String(500))
    criado_em = Column(DateTime, nullable=False, server_default=func.now())
    atualizado_em = Column(DateTime, nullable=False, server_default=func.now(), onupdate=func.now())

File: frontend/src/lib/dtos/prospection.ts
export class ProspectionRequest {
	readonly titulo: string;
	readonly segmento: string;
	readonly localizacao: string;
	readonly quantidade_leads: number;
	readonly prompt_customizado: string;
	readonly engine_tipo: string;

	constructor(data: Record<string, any>) {
		this.titulo = data.titulo ?? '';
		this.segmento = data.segmento ?? '';
		this.localizacao = data.localizacao ?? '';
		this.quantidade_leads = data.quantidade_leads ?? 10;
		this.prompt_customizado = data.prompt_customizado ?? '';
		this.engine_tipo = data.engine_tipo ?? 'claude_code';
	}

	isValid(): boolean {
		return this.titulo.trim().length > 0 && this.segmento.trim().length > 0;
	}

	toPayload(): Record<string, any> {
		return {
			titulo: this.titulo,
			segmento: this.segmento,
			localizacao: this.localizacao || null,
			quantidade_leads: this.quantidade_leads,
			prompt_customizado: this.prompt_customizado || null,
			engine_tipo: this.engine_tipo,
		};
	}
}

export interface ProspectionResumo {
	id: string;
	titulo: string;
	segmento: string;
	localizacao: string | null;
	engine_tipo: string;
	status: string;
	total_leads_encontrados: number;
	criado_em: string;
}

export interface ProspectionDetalhe {
	id: string;
	titulo: string;
	segmento: string;
	localizacao: string | null;
	quantidade_leads: number;
	prompt_customizado: string | null;
	engine_tipo: string;
	status: string;
	total_leads_encontrados: number;
	erro: string | null;
	arquivo_export_url: string | null;
	leads: LeadResumo[];
	criado_em: string;
	atualizado_em: string;
}

export interface LeadResumo {
	id: string;
	nome_empresa: string;
	segmento: string | null;
	website: string | null;
	score: number;
	status: string;
}
